napari_aicsimageio/core.py

```diff
@@ -41,6 +63,77 @@ def _get_full_image_data(img: AICSImage, in_memory: bool) -> Optional[xr.DataArr
     return None
 
 
+# Function to handle multi-scene files.
+def _get_scenes(img: AICSImage, in_memory: bool) -> Optional[xr.DataArray]:
+    # Create the list widget and populate with the scenes in the file
+    list_widget = QListWidget()
+    for scene in img.scenes:
+        list_widget.addItem(scene)
+    viewer = _get_viewer()
+    viewer.window.add_dock_widget([list_widget], area="right", name="Scene Selector")
+
+    # Function to create image layer from a scene selected in the list widget
+    def open_scene(item):
+        scene = item.text()
+        img.set_scene(scene)
+        if DimensionNames.MosaicTile in img.reader.dims.order:
+            try:
+                if in_memory:
+                    data = img.reader.mosaic_xarray_data
+                else:
+                    data = img.reader.mosaic_xarray_dask_data
+
+            # Catch reader does not support tile stitching
+            except NotImplementedError:
+                print(
+                    "AICSImageIO: Mosaic tile stitching "
+                    "not yet supported for this file format reader."
+                )
+        else:
+            if in_memory:
+                data = img.reader.xarray_data
+            else:
+                data = img.reader.xarray_dask_data
+        meta = _get_meta(data, img)
+        viewer.add_image(data, name=scene, metadata=meta, scale=meta["scale"])
+
+    list_widget.currentItemChanged.connect(open_scene)
+    return None
+
+
+# Function to get Metadata to provide with data
+def _get_meta(data, img):
+    meta = {}
+    if DimensionNames.Channel in data.dims:
+        # Construct basic metadata
+        meta["name"] = data.coords[DimensionNames.Channel].data.tolist()
+        meta["channel_axis"] = data.dims.index(DimensionNames.Channel)
+    # Not multi-channel, use current scene as image name
+    else:
+        meta["name"] = img.reader.current_scene
+    # Handle samples / RGB
+    if DimensionNames.Samples in img.reader.dims.order:
+        meta["rgb"] = True
+    # Handle scales
+    scale: List[float] = []
+    for dim in img.reader.dims.order:
+        if dim in [
+            DimensionNames.SpatialX,
+            DimensionNames.SpatialY,
+            DimensionNames.SpatialZ,
+        ]:
+            scale_val = getattr(img.physical_pixel_sizes, dim)
+            if scale_val is not None:
+                scale.append(scale_val)
+    # Apply scales
+    if len(scale) > 0:
+        meta["scale"] = tuple(scale)
+    # Apply all other metadata
+    meta["metadata"] = {"ome_types": img.metadata}
```

Comment:
Super nitpicky of me but can you add back the spacing / blank lines to this function. All the blank lines before comments were removed it looks like.

Reply:
Yeah, I think that might be a VS Code formatter thing. When I save it changes stuff—usually for the better though—but I should figure out the rules.